lib/wahwah/mp3_tag.rb

```diff
@@ -50,11 +50,21 @@ def parse
     end
 
     def parse_id3_tag
-      id3_v1_tag = ID3::V1.new(@file_io.dup)
-      id3_v2_tag = ID3::V2.new(@file_io.dup)
-
-      return id3_v2_tag if id3_v2_tag.valid?
-      id3_v1_tag if id3_v1_tag.valid?
+      @file_io.seek(0)
+      signature = @file_io.read(6)
+      @file_io.seek(0)
```

Comment:
No need to rewind twice, because ID3::V2 and ID3:V1 will seek to specific position when parse.